tests/mpls/test_mpls.py

```diff
@@ -0,0 +1,594 @@
+
+        exp_pkt = pkt.copy()
+        exp_pkt['MPLS'].ttl -= 1
+        exp_pkt['MPLS'].label = exp_label
+        exp_pkt = mask.Mask(exp_pkt)
+        exp_pkt.set_do_not_care_scapy(packet.Ether, 'dst')
+        exp_pkt.set_do_not_care_scapy(packet.Ether, 'src')
+        exp_pkt.set_do_not_care_scapy(packet.IP, 'chksum')
+
+        return exp_pkt
+    
+    def expected_mask_mpls_push_packet(self, pkt, exp_label):
+        """ return mask for mpls packet """
+        
+        exp_pkt = pkt.copy()
+        exp_pkt['MPLS'].ttl = exp_pkt['IP'].ttl - 1
+        exp_pkt['IP'].ttl -= 1
+        exp_pkt['MPLS'].label = exp_label
+        exp_pkt = mask.Mask(exp_pkt)
+        exp_pkt.set_do_not_care_scapy(packet.Ether, 'dst')
+        exp_pkt.set_do_not_care_scapy(packet.Ether, 'src')
+        exp_pkt.set_do_not_care_scapy(packet.IP, 'chksum')
+        
+        return exp_pkt
+
+    def test_pop_label(self, setup, ptfadapter):
+        """ test pop label """
+            
+        setup = setup
+        duthost = setup['duthost']
+
+        dst_pid=self.get_dst_portid(setup)
+        src_pid=self.get_src_portid(setup)
+
+        dst_port=setup['tor_ports'][dst_pid]
+        src_port=setup['spine_ports'][src_pid]
+
+        dst_peer_addr=setup['tor_peer_addr'][dst_pid]
+        src_peer_addr=setup['spine_peer_addr'][src_pid]
+
+        config_variables = {
+            'dst_port': dst_port,
+            'src_port': src_port,
+            'dst_peer_addr': dst_peer_addr,
+            'src_peer_addr': src_peer_addr,
+        }
+
+        logger.info('extra variables for MPLS config:\n{}'.format(pprint.pformat(config_variables)))
+        duthost.host.options['variable_manager'].extra_vars.update(config_variables)
+        
+        logger.info('generate config for MPLS')
+        mpls_config = '{}.json'.format(LABEL_POP_ROUTES)
+        mpls_config_path = os.path.join(setup['dut_tmp_dir'], mpls_config)
+        template_file='{}.j2'.format(LABEL_POP_ROUTES)
+        duthost.template(src=os.path.join(ADD_DIR, template_file), dest=mpls_config_path)
+
+        result = duthost.shell('config interface mpls add {}'.format(src_port),
+                           module_ignore_errors=True)
+        if result['rc'] != 0:
+            pytest.fail('Failed to apply labelroute configuration file: {}'.format(result['stderr']))
+
+        result = duthost.shell('config interface mpls add {}'.format(dst_port),
+                           module_ignore_errors=True)
+        if result['rc'] != 0:
+            pytest.fail('Failed to apply labelroute configuration file: {}'.format(result['stderr']))
+
+        # Apply routes with swssconfig
+        result = duthost.shell('docker exec -i swss swssconfig /dev/stdin < {}'.format(mpls_config_path),
+                           module_ignore_errors=True)
+        if result['rc'] != 0:
+            pytest.fail('Failed to apply labelroute configuration file: {}'.format(result['stderr']))
+            
```

Comment:
Can we use a fixture to generate config for MPLS ? These code are similar in each test case.

Reply:
Added config_interface_mpls